cmd/ig/daemon.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	gadgetservice "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/runtime"
+)
+
+func addDaemonCommand(rootCmd *cobra.Command, runtime runtime.Runtime) {
+	daemonCmd := &cobra.Command{
+		Use:   "daemon",
+		Short: "runs Inspektor Gadget as a daemon",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return nil
+		},
```

Comment:
use `SilenceUsage: true,`? Otherwise I get weird things like:

```bash
$ ./ig daemon 
Error: ig must be run as root to be able to run eBPF programs
Usage:
  ig daemon [flags]

Flags:
      --daemon-group-id int    Group id that the unix socket should use (if daemon-socket is set to e.g. unix:///path/to.socket)
      --daemon-socket string   The socket to listen on for new Inspektor Gadget requests. Can be a unix socket (unix:///path/to.socket) or a tcp socket (tcp://127.0.0.1:1234) (default "unix:///var/run/ig.socket")
  -h, --help                   help for daemon

Global Flags:
      --auto-mount-filesystems          Automatically mount bpffs, debugfs and tracefs if they are not already mounted
      --auto-sd-unit-restart            Automatically run in a privileged systemd unit if lacking enough capabilities
      --auto-wsl-workaround             Automatically find the host procfs when running in WSL2
      --containerd-namespace string     Containerd namespace to use (default "k8s.io")
      --containerd-socketpath string    Co......
```